agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/revoke/RevokingServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.dke.data.agrirouter.impl.revoke;
+
+import com.dke.data.agrirouter.api.dto.revoke.RevokeRequest;
+import com.dke.data.agrirouter.api.enums.RevokeResponse;
+import com.dke.data.agrirouter.api.env.Environment;
+import com.dke.data.agrirouter.api.service.onboard.RevokingService;
+import com.dke.data.agrirouter.api.service.parameters.RevokeParameters;
+import com.dke.data.agrirouter.impl.EnvironmentalService;
+import com.dke.data.agrirouter.impl.RequestFactory;
+import com.dke.data.agrirouter.impl.SignatureService;
+import com.dke.data.agrirouter.impl.common.UtcTimeService;
+import com.google.gson.Gson;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.commons.codec.binary.Hex;
+
+public class RevokingServiceImpl extends EnvironmentalService
+    implements RevokingService, SignatureService {
+
+  public RevokingServiceImpl(Environment environment) {
+    super(environment);
+  }
+
+  @Override
+  public RevokeResponse revoke(RevokeParameters revokeParameters) {
+    revokeParameters.validate();
+
+    RevokeRequest revokeRequest = createRevokeRequestBody(revokeParameters);
+    Gson gson = new Gson();
+    String jsonBody = gson.toJson(revokeRequest).replace("\n", "");
+    String encodedSignature = this.createSignature(revokeParameters, jsonBody);
+    this.verifySignature(jsonBody, decodeHex(encodedSignature), revokeParameters.getPublicKey());
+
+    Response response =
+        RequestFactory.signedDeleteRequest(
+                environment.getRevokeUrl(), revokeParameters.getApplicationId(), encodedSignature)
+            .build("DELETE", Entity.entity(jsonBody, MediaType.APPLICATION_JSON_TYPE))
+            .invoke();
+
+    if (response.getStatus() == RevokeResponse.SUCCESS.getKey()) {
+      return RevokeResponse.SUCCESS;
+    } else if (response.getStatus() == RevokeResponse.UNAUTHORIZED.getKey()) {
+      return RevokeResponse.UNAUTHORIZED;
+    } else if (response.getStatus() == RevokeResponse.VALIDATION_ERROR.getKey()) {
+      return RevokeResponse.VALIDATION_ERROR;
+    } else {
+      return null;
```

Comment:
`return null` is bad habbit. Either return a valid response or throw an exception of the response could not be detected.

Reply:
Fixed